paper.md

```diff
@@ -2,11 +2,77 @@
+    Researchers working in fields of computational science often find it difficult to reproduce experiments from artifacts like code, data, diagrams and results which are left behind by the previous researchers. 
+    The code developed on one machine often fails to run on other machines due to differences in hardware architecture, OS, software dependencies, among others. 
+    This is accompanied by the difficulty in understanding how artifacts are organized, as well as in using them in correct order. 
+    Since this problem of platform dependency can be solved by using Software containers (light weight virtualization), researchers and developers have built scientific workflow engines like Argo, Pachyderm, Brigade, etc. for organizing the steps in a workflow as Directed Acyclic Graphs and running them in separate containers.
+    But these existing container-native workflow engines assume the availability of a Kubernetes cluster deployed in the cloud, access to which is not always trivial.
+    Therefore, there is a need for a container-native workflow engine that does not assume the presence of a Kubernetes cluster or any other specific computing environment.
+    In this paper, we introduce Popper, a container-native workflow engine that executes each step of a workflow in a separate dedicated container and does not assume the presence of a Kubernetes cluster or any cloud based Kubernetes service.
+    With Popper, researchers can build and validate workflows easily in almost any environment of their choice like a local machine, a Slurm based HPC cluster or a Kubernetes based cloud computing environment. 
+    To exemplify the suitability of this workflow engine, we present three case studies where we take examples from Machine Learning and HPC and turn them into Popper workflows.
 ---
 
 # Introduction {#sec:intro}
 
+<!-- defining the problem of reproducibility in computational science -->
+
+Around 48.6% of scientists and researchers working in various domains related to computational science, upload experimental artifacts like code, figures, datasets, configuration files, etc. on open-access repositories like Zenodo [@_zenodo_], Figshare [@_figshare_] or GitHub [@github]. 
+Unfortunately, only 1.1% of the artifacts available online are fully reproducible and 0.6% of them are partially reproducible [@stagge2019assessing]. 
+According to a study of 2016 by Nature, among a group of 1576 scientists around 70% of them failed to reproduce each other's experiments [@baker2016reproducibility].
+This problem occurs mostly due to the lack of proper documentation, missing artifacts, broken software dependencies, etc. 
+Consequently, this results in other researchers wasting time trying to figure out how to reproduce those experiments from the archived artifacts. 
+Hence, it makes the process inefficient and cumbersome leading to what is known as the Reproducibility crisis or Replication crisis in science [@sep-scientific-reproducibility].
```

Comment:
We can remove the chained implication (Consequently, ... , Hence) by merging sentences. Sometimes we can omit these transitional words and the message is still communicated properly. Like:

> This problem occurs mostly due to... This results in other researchers wasting time..., ultimately making this process inefficient, cumbersome and error prone [@sep-scientific-reproducibility].

We can also point to the reference directly without explicitly naming the problem (as I did above).

Reply:
This reads real good.